crypto/elliptic-curve-crypto/src/lib.rs

```diff
@@ -62,70 +63,72 @@ lazy_static! {
     };
 }
 
-// TODO (SECURITY): Use side-channel-resistant math
-pub fn private_to_public(private_key: &U256) -> Affine {
-    Affine::from(&base_mul(&*GENERATOR_TABLE, private_key % ORDER))
+// TODO (SECURITY): The signatures are malleable in w -> -w.
+#[derive(PartialEq, Eq, Clone, Hash)]
+#[cfg_attr(feature = "std", derive(Debug))]
+pub struct Signature {
+    r: ScalarFieldElement,
+    w: ScalarFieldElement,
 }
 
-fn divmod(a: &U256, b: &U256) -> Option<U256> {
-    b.inv_mod(&ORDER).map(|bi| a.mul_mod::<Order>(&bi))
+// TODO (SECURITY): Use side-channel-resistant math
+pub fn private_to_public(private_key: &ScalarFieldElement) -> Affine {
+    Affine::from(&base_mul(&*GENERATOR_TABLE, private_key.to_uint()))
```

Comment:
It makes sense to change the arguments of `base_mul` and it's sibblings to use `ScalarFieldElement` as well.